cli/flox/src/commands/environment.rs

```diff
@@ -421,18 +422,47 @@ impl Activate {
         let now_active =
             UninitializedEnvironment::from_concrete_environment(&concrete_environment)?;
 
-        let mut environment = concrete_environment.into_dyn_environment();
+        let environment = concrete_environment.dyn_environment_ref_mut();
 
         // Don't spin in bashrcs and similar contexts
-        let activation_path = if !stdout().is_tty() && self.run_args.is_empty() {
-            environment.activation_path(&flox)?
+        let activation_path_result = if !stdout().is_tty() && self.run_args.is_empty() {
+            environment.activation_path(&flox)
         } else {
             Dialog {
                 message: &format!("Getting ready to use environment {now_active}..."),
                 help_message: None,
                 typed: Spinner::new(|| environment.activation_path(&flox)),
             }
-            .spin()?
+            .spin()
+        };
+
+        let activation_path = match activation_path_result {
+            Err(
+                err @ EnvironmentError2::Core(CoreEnvironmentError::LockedManifest(
+                    LockedManifestError::BuildEnv(CallPkgDbError::PkgDbError(PkgDbError {
+                        exit_code: 100,
+                        ..
+                    })),
+                )),
+            ) => {
+                let mut message = formatdoc! {"
+                    ❌ This environment is not yet compatible with your system ({system}).
+
+                    {err}",
+                    system = flox.system
+                };
+
+                if let ConcreteEnvironment::Remote(remote) = &concrete_environment {
+                    message.push_str("\n\n");
+                    message.push_str(&format!(
+                    "Use 'flox pull --amend-system {}/{}' to update and verify this environment on your system.",
```

Comment:
```suggestion
                    "Use 'flox pull --add-system {}/{}' to update and verify this environment on your system.",
```

see [rename amend-system -> add-system](https://github.com/flox/flox/pull/834/commits/71875282505102aef90ead9f02358e1b0f120589)